packages/opentelemetry-instrumentation-anthropic/opentelemetry/instrumentation/anthropic/__init__.py

```diff
@@ -220,27 +251,23 @@ async def _aset_token_usage(
     if usage := response.get("usage"):
         prompt_tokens = usage.input_tokens
     else:
-        prompt_tokens = 0
-        if hasattr(anthropic, "count_tokens"):
-            if request.get("prompt"):
-                prompt_tokens = await anthropic.count_tokens(request.get("prompt"))
-            elif request.get("messages"):
-                prompt_tokens = 0
-                for m in request.get("messages"):
-                    content = m.get("content")
-                    if isinstance(content, str):
-                        prompt_tokens += await anthropic.count_tokens(content)
-                    elif isinstance(content, list):
-                        for item in content:
-                            # TODO: handle image tokens
-                            if isinstance(item, dict) and item.get("type") == "text":
-                                prompt_tokens += await anthropic.count_tokens(
-                                    item.get("text", "")
-                                )
-
-    if token_histogram and type(prompt_tokens) is int and prompt_tokens >= 0:
+        prompt_tokens = await acount_prompt_tokens_from_request(anthropic, request)
+
+    if usage := response.get("usage"):
+        cache_read_tokens = dict(usage).get("cache_read_input_tokens", 0)
+    else:
+        cache_read_tokens = 0
+
+    if usage := response.get("usage"):
+        cache_creation_tokens = dict(usage).get("cache_creation_input_tokens", 0)
+    else:
+        cache_creation_tokens = 0
+
+    input_tokens = prompt_tokens + cache_read_tokens + cache_creation_tokens
+
+    if token_histogram and type(input_tokens) is int and input_tokens >= 0:
         token_histogram.record(
-            prompt_tokens,
+            input_tokens,
```

Comment:
I am open to discussing whether this must be input tokens (i.e. all tokens sent by the user) or prompt tokens (i.e. the new tokens that anthropic has neither written to nor read from cache)